SpatialGDK/Source/SpatialGDKServices/Private/SSpatialOutputLog.h

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "Developer/OutputLog/Private/SOutputLog.h"
+#include "HAL/FileManagerGeneric.h"
+#include "IDirectoryWatcher.h"
+#include "SlateFwd.h"
+
+DECLARE_LOG_CATEGORY_EXTERN(LogSpatialOutputLog, Log, All);
+
+// Child class of the file reader used by Unreal but with the ability to update the known file size.
+// This allows us to read a log file while it is being written to.
+class FArchiveLogFileReader : public FArchiveFileReaderGeneric
```

Comment:
Does FArchiveLogFileReader definition need to be in header file?

Reply:
Investigated together. Can't forward declare the `FArchiveLogFileReader` for use in other methods in this class when it's being used as a `TUniquePtr`